### DISCUSSION TITLE
FeatureRequest: Grafana 7 - Transform / Reduce / Calculation -> Add statistic calculations like "Standard deviation" ....

### DISCUSSION THREAD

**Discussion by Bubbu17** [grafana/grafana#38381] [Dashboards, Panels & Visualizations]
**What would you like to be added**:
Hello ,

I realy appreciate the new "Transform" feature section in Grafana 7 but i still miss Statistic functions in the section "Reduce" / " Calculate" like :
Parameters

Population mean = μ = ( Σ Xi ) / N
Population standard deviation = σ = sqrt [ Σ ( Xi - μ )2 / N ]
Population variance = σ2 = Σ ( Xi - μ )2 / N
Variance of population proportion = σP2 = PQ / n
Standardized score = Z = (X - μ) / σ
Population correlation coefficient = ρ = [ 1 / N ] * Σ { [ (Xi - μX) / σx ] * [ (Yi - μY) / σy ] }

Statistics

Unless otherwise noted, these formulas assume simple random sampling.

Sample mean = x = ( Σ xi ) / n
Sample standard deviation = s = sqrt [ Σ ( xi - x )2 / ( n - 1 ) ]
Sample variance = s2 = Σ ( xi - x )2 / ( n - 1 )
Variance of sample proportion = sp2 = pq / (n - 1)
Pooled sample proportion = p = (p1 * n1 + p2 * n2) / (n1 + n2)
Pooled sample standard deviation = sp = sqrt [ (n1 - 1) * s12 + (n2 - 1) * s22 ] / (n1 + n2 - 2) ]
Sample correlation coefficient = r = [ 1 / (n - 1) ] * Σ { [ (xi - x) / sx ] * [ (yi - y) / sy ] }

![image](https://user-images.githubusercontent.com/62136070/82822742-92085200-9ea6-11ea-9686-6cdce56068db.png)

**Why is this needed**:
It is needed to do further statistic analysis on e.g. Production data on e.g. what is the  Standard deviation of a certain value....

**Comment by pkolyvas** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Comment by ryantxu** [comment]
Added in 9.1 https://github.com/grafana/grafana/pull/52769

**Comment by kk6mrp** [comment]
Hi! It would be super useful to add correlation coefficient to the legend value calculation options as well as the Transform -> Reduce options in order to better understand if trending over time!

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
